tests/testthat/test_projects.R

```diff
@@ -1,29 +1,41 @@
-test_that("'.index_path()' works for projects", {
-    expect_equal(
-        .index_path(filters = URLencode("{}", reserved = TRUE),
-                    size = 10L,
-                    sort = "projectTitle",
-                    order = "asc",
-                    catalog = "dcp2",
-                    base_path = "/index/projects"),
-        "/index/projects?filters=%7B%7D&size=10&sort=projectTitle&order=asc&catalog=dcp2"
+test_that("'projects()' works with default arguments", {
+    test_proj <- projects()
+    expect_true(tibble::is_tibble(test_proj))
+    expect_true(nrow(test_proj) > 0L)
+    expect_equal(names(attr(test_proj, "columns")),
+                 projects_default_columns("tibble")$name)
+})
+
+test_that("'projects()' works for projectTitles with special characters", {
+    test_proj <- projects(
+        filters = filters(projectTitle = list(is = "Precursors of human CD4+ cytotoxic T lymphocytes identified by single-cell transcriptome analysis"))
     )
+    expect_equal(test_proj$projectId[[1]],
+                 "90bd6933-40c0-48d4-8d76-778c103bf545")
 })
 
-test_that("'.index_GET()' works for projects", {
+test_that("'projects()' works for projectTitles with line breaks", {
+    test_proj <- projects(
```

Comment:
This is an interesting test. In some ways one would like to 'fail early'. After the fact we know this will fail because the  `is = ` argument of the filter contains spaces. So should filter construction have checked to see that these values don't have line breaks? But maybe some values *can* have line breaks?

Reply:
I agree. I attempted to strip the line breaks from a title before encoding it but that changed the final encoding in such a way that the url was no longer valid. Would we just want to implement a `stopifnot()` for values containing line breaks?

Reply:
yes, `stopifnot()` sounds like the way to go...